forge-game/src/main/java/forge/game/trigger/TriggerAbilityTriggered.java

```diff
@@ -87,6 +87,20 @@ public final boolean performTest(final Map<AbilityKey, Object> runParams) {
                 return false;
             }
         }
+        
+        if (hasParam("TriggeredOwnAbility") && "True".equals(getParam("TriggeredOwnAbility")))
+        {
+            boolean match = false;
```

Comment:
shouldn't that be better with `Iterables.contains(causes, source)`?

also the part above with "ValidCause" can be shorted to just `matchesValidParam("ValidCause", causes)`

it is smart enough to handle iterables